ci/scripts/goLint.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+echo "installing Go lint..."
+go get golang.org/x/lint
+
+golint -set_exit_status ./...
+
+if [ $? -gt 0 ]
+  then
+    echo "Go linter returned errors"
+    exit 1
+  else
+    echo "golint did not identify any issues"
+    exit 0
+fi
```

Comment:
`golint` has been deprecated and is no longer supporter. I also had issues as the lint binary was not available within the Docker image.

I've left it in the PR to demonstrate a way we could implement it if its something we think is valuable.